### ISSUE TITLE
skipif marker ignored in parametrized list (2.9.0-rc.1)

### ISSUE THREAD

**Issue by nicoddemus** [pytest-dev/pytest#1412]
Consider this example:

```python
import pytest, sys

@pytest.mark.parametrize('x', [
    0,
    1,
    pytest.mark.skipif('True')(2)
])
def testSkipIf(x):
    assert x < 2    
```

In `2.8.7` it works as expected, the last parameter is skipped:

```
] py.test -q foo.py
..s
2 passed, 1 skipped, 1 pytest-warnings in 0.01 seconds
```

In `2.9.0-rc.1` though the test executes anyway, ignoring the `skipif` marker:

```
] py.test -q foo.py
..F
================================== FAILURES ===================================
________________________________ testSkipIf[2] ________________________________

x = 2

    @pytest.mark.parametrize('x', [
        0,
        1,
        pytest.mark.skipif('True')(2)
    ])
    def testSkipIf(x):
>       assert x < 2
E       assert 2 < 2

foo.py:9: AssertionError
1 failed, 2 passed, 1 pytest-warnings in 0.02 seconds
```

**Comment by nicoddemus** [comment]
A `git biscect` run finds this commit as bad:

```
9e57954b0388d2830dee1e560347af581b75534a is the first bad commit
```

**Comment by The-Compiler** [comment]
Link for the lazy: https://github.com/pytest-dev/pytest/commit/9e57954b0388d2830dee1e560347af581b75534a

**Comment by nicoddemus** [comment]
@The-Compiler thanks! 

Is there any chance the failing tests in `qutebrowser` are related to this? A quick check is to see if any of the failing tests are using `skipif` or `xfail` in a parametrized list. :grin: 

Either way I plan to at least take a look in `qutebrowser` later tonight.

**Comment by The-Compiler** [comment]
- With the current master (310bada): 17 failed
- When I revert 9e57954b0388d2830dee1e560347af581b75534a: 12 failed

So it seems this commit is *one* issue, but there are probably others

**Comment by nicoddemus** [comment]
Hmm, OK, thanks! I will take a closer look later.

**Comment by The-Compiler** [comment]
Seems like I messed up something with the revert (as there were conflicts), as I managed to trace another issue to 9e57954b0388d2830dee1e560347af581b75534a as well.

This test gets run instead of skipped:

```python
import pytest

@pytest.mark.skipif(True, reason='skip test')
def test_foo():
    pass
```

**Comment by The-Compiler** [comment]
cc @MichaelAquilina

**Comment by The-Compiler** [comment]
Okay, I reverted it again and resolved the conflict to the old/left side in #1109 - then all my tests pass.
